### ISSUE TITLE
[BUG] Unable to Build an Azure ML Container Image for an MLflow model

### ISSUE THREAD

**Issue by manish-globant** [mlflow/mlflow#5579]
I am unable to build an Azure ML Container Image for an MLflow model. While building the docker image using the code:

azure_image, azure_model = mlflow.azureml.build_image(model_uri="<model_uri path>", workspace=<azure_workspace>, synchronous=True)

Getting the following error in the log:

Step 9/13 : RUN pip install mlflow==1.24.0
 ---> Running in c52dae9d410a
[91mERROR: Could not find a version that satisfies the requirement mlflow==1.24.0 (from versions: 0.0.1, 0.1.0, 0.2.0, 0.2.1, 0.3.0, 0.4.0, 0.4.1, 0.4.2, 0.5.0, 0.5.1, 0.5.2, 0.6.0, 0.7.0, 0.8.0, 0.8.1, 0.8.2, 0.9.0, 0.9.0.1, 0.9.1, 1.0.0, 1.1.0, 1.1.1.dev0, 1.2.0, 1.3.0, 1.4.0, 1.5.0, 1.6.0, 1.7.0, 1.7.1, 1.7.2, 1.8.0, 1.9.0, 1.9.1, 1.10.0, 1.11.0, 1.12.0, 1.12.1, 1.13, 1.13.1, 1.14.0, 1.14.1, 1.15.0, 1.16.0, 1.17.0, 1.18.0, 1.19.0, 1.20.0, 1.20.1, 1.20.2, 1.21.0, 1.22.0, 1.23.0, 1.23.1)
[0m[91mERROR: No matching distribution found for mlflow==1.24.0

Also, I noticed that the python image being used is a python-slim while building the image:

Step 1/13 : FROM mcr.microsoft.com/azureml/o16n-base/python-slim:latest.westeurope
latest.westeurope: Pulling from azureml/o16n-base/python-slim

Could that also be a reason. If yes, how do we specify the python version while building the image?

**Comment by harupy** [comment]
Are you using python 3.6? 1.24.0 only supports python >= 3.7.

**Comment by manish-globant** [comment]
> Are you using python 3.6? 1.24.0 only supports python >= 3.7.

My python version is 3.9.7

**Comment by harupy** [comment]
In which python version was your model logged?

**Comment by manish-globant** [comment]
> In which python version was your model logged?

3.8

**Comment by harupy** [comment]
```
FROM mcr.microsoft.com/azureml/o16n-base/python-slim:latest.westeurope

RUN python --version
```

This dockerfile gave:

```
...
#5 [2/2] RUN python --version
#5 sha256:91cce686a547879f8385f286ce69093feebe40a90eb550aa049bbcf7d34e37d0
#5 0.274 Python 3.6.10 :: Anaconda, Inc.
#5 DONE 1.2s
```

**Comment by manish-globant** [comment]
So the python version is 3.6 in the image. Is there any way to specify the python version while building this image?

**Comment by manish-globant** [comment]
Ok, sure

**Comment by harupy** [comment]
@manishpyy `ContainerImage.image_configuration` provides a parameter named `base_image` which specifies a base image to be used:

https://docs.microsoft.com/en-us/python/api/azureml-core/azureml.core.image.containerimage?view=azure-ml-py#azureml-core-image-containerimage-image-configuration

However, MLflow doesn't expose this parameter.

**Comment by harupy** [comment]
btw `mlflow.azureml.build_image` is deprecated and we'd recommend using the azureml deployment plugin: https://aka.ms/aml-mlflow-deploy. Could you give this a shot?

**Comment by manishpyy** [comment]
Yes I did try the get_deploy_client api from the mlflow.deployments
from mlflow.deployments import get_deploy_client

I guess the above is possible only if the MLflow tracking URI is set to the Azure ML Workspace. However in my case, we need the mlflow server on another external server, and because of it I'm getting an error:

MlflowException: No plugin found for managing model deployments to "http". In order to deploy models to "http", find and install an appropriate plugin from https://mlflow.org/docs/latest/plugins.html#community-plugins using your package manager (pip, conda etc).

Is there any way to have the MLflow URI other than Azure ML Workspace.
